### PR TITLE
Dockerfile: adding a number of features (see README).

### PR DESCRIPTION
**Pull request by karthiksn**
See README.md for details.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,36 +1,93 @@
...
+```
+6. To create and connect to your Docker container, simply run the above command.
+
+7. To run Adept:
+```
+cd clients; git clone https://github.com/heronsystems/adeptRL
 cd adeptRL
-sudo docker build -t adept -f ./docker/Dockerfile .
+sudo python setup.py develop
+python -m adept.scripts.local ActorCritic
```
Comment (jtatusko) [inline docker/README.md:47]: i don't think you need to specify ActorCritic, should default to that

File: docker/connect.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    )
+    return parser.parse_args()
+
+
+def runcmd(cmd):
+    print("Running --\n{}\n--".format(cmd))
+    return os.system(cmd)
+
+
+def connectLocal(args):
```
Comment (jtatusko) [inline docker/connect.py:51]: python style method naming:
`def connect_local(args):`

File: docker/connect.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    try:
+        import pyperclip
+
+        pyperclip.copy(cmd)
+    except:
+        pass
+
+
+def main():
+    connectLocal(parse_args())
```
Comment (jtatusko) [inline docker/connect.py:127]: same

File: docker/connect.py (hunk 2 above)
Comment (karthiksn) [reply docker/connect.py:51]: Good catch.

File: docker/Dockerfile [hunk 4]
```diff
@@ -1,33 +1,225 @@
...
-# Set Python 3 as the default
-RUN ln -s /usr/bin/python3 /usr/bin/python
-
-# Install AdeptRL
-COPY ./ /adeptRL
-WORKDIR /adeptRL
-RUN pip3 install .[all]
+# Copyright (C) 2020 Heron Systems, Inc.
+#
+# This Dockerfile builds an image that:
```
Comment (jtatusko) [inline docker/Dockerfile:3]: from here down should be a docstring, not a comment

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,33 +1,225 @@
...
+        scipy==1.3.1 \
+        six==1.12.0 \
+        scikit-learn==0.20.1 \
+        scikit-optimize==0.7.4 \
+        tabulate==0.8.7 \
+        tensorboard==1.14.0 \
+        tensorflow==1.14.0 \
+        tensorboardX==2.0 \
+        threadpoolctl==2.0.0 \
+        torch==1.4.0 \
```
Comment (jtatusko) [inline docker/Dockerfile:129]: i believe 1.5.0 is out

File: docker/README.md [hunk 6]
```diff
@@ -1,36 +1,93 @@
...
+```
+python connect.py --dockerfile /path/to/your/Dockerfile --username karthik --email karthik@starfruit-llc.com --fullname "Karthik Narayan"
+```
+
+3. This command should fail the first time you run, asking you to create a few directories and symlink your `.ssh` directory. You may get a message similar to the following:
+
+```
+Please run: mkdir -p /mnt/users/karthik and run: chmod -R a+rw /mnt/users/karthik. You may need to run these commands using sudo.
+Please run the following before running connect.py. This is to allow your Docker instance to be able to pull/push from private github repositories:
+ln -s ~/.ssh /mnt/users/karthik/.ssh
```
Comment (jtatusko) [inline docker/README.md:31]: ln takes target first:
`ln -s /mnt/users/karthik/.ssh ~/.ssh`

Comment (karthiksn) [reply docker/README.md:31]: Nice catch!

File: docker/Dockerfile (hunk 4 above)
Comment (jtatusko) [reply docker/Dockerfile:3]: :facepalm:

Comment (karthiksn) [reply docker/Dockerfile:3]: N/A - spoke offline; this isn't a python file.

**Comment by karthiksn** [comment]
FYI - I just upgraded Docker on my machine and can confirm that the latest commit (with the updated `connect.py`) works without `nvidia-docker`.

File: docker/Dockerfile [hunk 7]
```diff
@@ -1,33 +1,225 @@
...
+    mkdir -p /opt/cudnn7 && \
+    cd /opt/cudnn7/ && \
+    wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb && \
+    dpkg -i nvidia-machine-learning-repo-ubuntu1804_1.0.0-1_amd64.deb && \
+    apt-get update && \
+    $APT_INSTALL libcudnn7-dev && \
+    ldconfig && \
+    apt-get clean && \
+    apt-get autoremove && \
+    rm -rf /var/lib/apt/lists/* /tmp/* ~/*
```
Comment (benbellheron) [inline docker/Dockerfile:150]: Unneeded, just use the cudnn runtime from nvidia like: FROM nvidia/cuda:10.2-cudnn7-runtime-ubuntu18.04

Comment (karthiksn) [reply docker/Dockerfile:150]: We may want to consider `FROM nvidia/cudagl:10.1-devel-ubuntu18.04`, as this allows users to invoke and view OpenGL-based rendering applications from within the Docker container.

I'm not aware of a base that supports OpenGL-based rendering + also has cuDNN.

File: docker/Dockerfile [hunk 8]
```diff
@@ -1,33 +1,225 @@
...
+        scikit-learn==0.20.1 \
+        scikit-optimize==0.7.4 \
+        tabulate==0.8.7 \
+        tensorboard==1.14.0 \
+        tensorflow==1.14.0 \
+        tensorboardX==2.0 \
+        threadpoolctl==2.0.0 \
+        torch==1.5.0 \
+        torchvision==0.6.0 \
+        tqdm==4.43.0
```
Comment (benbellheron) [inline docker/Dockerfile:131]: should be in a requirements.txt file if these are adept deps.

**Pull request merged**